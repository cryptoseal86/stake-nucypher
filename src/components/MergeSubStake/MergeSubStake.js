import React, { useState } from 'react';
import { Form, Col, Row, Button } from 'react-bootstrap';
import { Formik } from 'formik';
import Loading from '../Loading/Loading';
import * as Yup from 'yup';

import './MergeSubStake.scss';

const validationSchema = Yup.object().shape({
  mergeIdValue: Yup.number().required('Field is required.')
});

function MergeSubStake(props) {
  const onSubStakeMerge = (stakeValues) => {
    if (props.onSubStakeMerge) {
      props.onSubStakeMerge(stakeValues);
    }
  };
  const [loading, setLoading] = useState(false);
  const onSubmit = async (...args) => {
    setLoading(true);
    if (onSubStakeMerge) {
      await onSubStakeMerge(...args);
    }
    setLoading(false);
  };
  return <Formik onSubmit={onSubmit} initialValues={{ mergeIdValue: 0 }} validationSchema={validationSchema}>
    {({
      handleSubmit,
      handleChange,
      handleBlur,
      values,
      touched,
      isValid,
      errors,
      setValues,
      setErrors
    }) => {
      if (!loading) {
        return <Form onSubmit={handleSubmit}>
          <Form.Group as={Row}>
            <Form.Label column md={{ span: 2, offset: 2 }} htmlFor="substake-merge-id">
              Merge with #
            </Form.Label>
            <Col md={4}>
              <Form.Control id="substake-merge-id" type="number" name="prolongMergeValue" onChange={handleChange} value={values.prolongDurationValue} />
              <div className="feedback-placeholder">
                <Form.Control.Feedback type="invalid" style={{ display: errors.mergeIdValue && touched.mergeIdValue ? 'inline' : 'none' }}>{errors.mergeIdValue}</Form.Control.Feedback>
              </div>
            </Col>
          </Form.Group>
          <div className="d-flex justify-content-center">
            <Button type="submit"  disabled={!isValid}>Merge</Button>
          </div>
        </Form>;
      } else {
        return <>
          <div className="mt-2 d-flex justify-content-center align-items-center">
            <Loading size={60}></Loading>
          </div>
        </>;
      }
    }
  }
  </Formik>;
}

export default MergeSubStake;
