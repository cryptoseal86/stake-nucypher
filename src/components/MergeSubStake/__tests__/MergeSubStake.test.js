import React from 'react';
import MergeSubStake from '../MergeSubStake';
import { render, fireEvent, wait } from '@testing-library/react'

describe('<MergeSubStake />', () => {

  it('should render correctly', () => {
    const { asFragment } = render(<MergeSubStake />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('should trigger onSubStakeMerge on merge', async () => {
    const onSubStakeMerge = jest.fn(() => {});
    const { getByLabelText, getByText } = render(<MergeSubStake onSubStakeMerge={onSubStakeMerge} />);

    await wait(() => {
      fireEvent.change(getByLabelText('Merge with #'), { target: { value: '0' } });
      fireEvent.click(getByText('Merge'));
    });

    expect(onSubStakeMerge).toBeCalledWith(expect.objectContaining({ mergeIdValue: 0 }));
  });
});
