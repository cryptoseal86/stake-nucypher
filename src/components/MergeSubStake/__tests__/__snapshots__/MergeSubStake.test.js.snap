// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MergeSubStake /> should render correctly 1`] = `
<DocumentFragment>
  <form
    class=""
  >
    <div
      class="form-group row"
    >
      <label
        class="form-label col-form-label col-md-2 offset-md-2"
        for="substake-merge-id"
      >
        Merge with #
      </label>
      <div
        class="col-md-4"
      >
        <input
          class="form-control"
          id="substake-merge-id"
          name="prolongMergeValue"
          type="number"
          value=""
        />
        <div
          class="feedback-placeholder"
        >
          <div
            class="invalid-feedback"
            style="display: none;"
          />
        </div>
      </div>
    </div>
    <div
      class="d-flex justify-content-center"
    >
      <button
        class="btn btn-primary"
        type="submit"
      >
        Merge
      </button>
    </div>
  </form>
</DocumentFragment>
`;
